pkgs/tools/wayland/hyprpaper/default.nix

```
@@ -0,0 +1,56 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, pkg-config
+, libjpeg
+, mesa
+, pango
+, wayland
+, wayland-protocols
+, wayland-scanner
+, wlroots
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hyprpaper";
+  version = "unstable-2022-07-04";
+
+  src = fetchFromGitHub {
+    owner = "hyprwm";
+    repo = pname;
+    rev = "e15912e9817d79bb988085c88e313fac5ab60940";
+    sha256 = "sha256-UZSRcj+CckUDllBtmlIcwA+xXUonpJZl3zC151IV3f0=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    wayland-scanner
+  ];
```

```suggestion
, wlroots
, gitMinimal
}:

stdenv.mkDerivation rec {
  pname = "hyprpaper";
  version = "unstable-2022-07-04";

  src = fetchFromGitHub {
    owner = "hyprwm";
    repo = pname;
    rev = "e15912e9817d79bb988085c88e313fac5ab60940";
    leaveDotGit = true;
    sha256 = "sha256-1P4U6fSaiiUYAvoL8EdFPBbtLh5v/S2RMSuSpbISGFc=";
  };

  nativeBuildInputs = [
    cmake
    pkg-config
    wayland-scanner
    gitMinimal
  ];
```

With this change `hyprpaper` can know which commit it was built.
(Yes I calculated the hash for you, you can always re-calculate the hash again by removing the `sha256` line.)